pkg/runtime-tools/generate/generate_suite_test.go

```diff
@@ -0,0 +1,354 @@
+			spec.Linux.Resources = &rspec.LinuxResources{}
+		}
+		if spec.Linux.Resources.Memory == nil {
+			spec.Linux.Resources.Memory = &rspec.LinuxMemory{}
+		}
+		spec.Linux.Resources.Memory.Limit = &v
+	}
+}
+
+func withMemorySwap(v int64) specOption {
+	return func(spec *rspec.Spec) {
+		if spec.Linux == nil {
+			spec.Linux = &rspec.Linux{}
+		}
+		if spec.Linux.Resources == nil {
+			spec.Linux.Resources = &rspec.LinuxResources{}
+		}
+		if spec.Linux.Resources.Memory == nil {
+			spec.Linux.Resources.Memory = &rspec.LinuxMemory{}
+		}
+		spec.Linux.Resources.Memory.Swap = &v
+	}
+}
+
+func withCPUShares(v uint64) specOption {
+	return func(spec *rspec.Spec) {
+		if spec.Linux == nil {
+			spec.Linux = &rspec.Linux{}
+		}
+		if spec.Linux.Resources == nil {
+			spec.Linux.Resources = &rspec.LinuxResources{}
+		}
+		if spec.Linux.Resources.CPU == nil {
+			spec.Linux.Resources.CPU = &rspec.LinuxCPU{}
+		}
+		spec.Linux.Resources.CPU.Shares = &v
+	}
+}
+
+func withCPUQuota(v int64) specOption {
+	return func(spec *rspec.Spec) {
+		if spec.Linux == nil {
+			spec.Linux = &rspec.Linux{}
+		}
+		if spec.Linux.Resources == nil {
+			spec.Linux.Resources = &rspec.LinuxResources{}
+		}
+		if spec.Linux.Resources.CPU == nil {
+			spec.Linux.Resources.CPU = &rspec.LinuxCPU{}
+		}
+		spec.Linux.Resources.CPU.Quota = &v
+	}
+}
+
+func withCPUPeriod(v uint64) specOption {
+	return func(spec *rspec.Spec) {
+		if spec.Linux == nil {
+			spec.Linux = &rspec.Linux{}
+		}
+		if spec.Linux.Resources == nil {
+			spec.Linux.Resources = &rspec.LinuxResources{}
+		}
+		if spec.Linux.Resources.CPU == nil {
+			spec.Linux.Resources.CPU = &rspec.LinuxCPU{}
+		}
+		spec.Linux.Resources.CPU.Period = &v
+	}
+}
+
+func withCPUSetCPUs(v string) specOption {
+	return func(spec *rspec.Spec) {
+		if spec.Linux == nil {
+			spec.Linux = &rspec.Linux{}
+		}
+		if spec.Linux.Resources == nil {
+			spec.Linux.Resources = &rspec.LinuxResources{}
+		}
+		if spec.Linux.Resources.CPU == nil {
+			spec.Linux.Resources.CPU = &rspec.LinuxCPU{}
+		}
+		spec.Linux.Resources.CPU.Cpus = v
+	}
+}
+
+func withCPUSetMems(v string) specOption {
+	return func(spec *rspec.Spec) {
+		if spec.Linux == nil {
+			spec.Linux = &rspec.Linux{}
+		}
+		if spec.Linux.Resources == nil {
+			spec.Linux.Resources = &rspec.LinuxResources{}
+		}
+		if spec.Linux.Resources.CPU == nil {
+			spec.Linux.Resources.CPU = &rspec.LinuxCPU{}
+		}
+		spec.Linux.Resources.CPU.Mems = v
+	}
+}
+
+func makeSpec(options ...specOption) *rspec.Spec {
+	spec := &rspec.Spec{
+		Process: &rspec.Process{},
+		Linux: &rspec.Linux{
+			Resources: &rspec.LinuxResources{
+				Memory: &rspec.LinuxMemory{
+					Limit: Int64(12345),
+				},
+				CPU: &rspec.LinuxCPU{
+					Shares: Uint64(45678),
+					Quota:  Int64(87654),
+					Period: Uint64(54321),
+					Cpus:   "0-111",
+					Mems:   "0-4",
+				},
+			},
+		},
+	}
+	for _, o := range options {
+		o(spec)
+	}
+	return spec
+}
+
+func Int64(v int64) *int64 {
```

Comment:
I think we can use generic here, like 

```
func ToPtr[T any](v T) *T {
    return &v
}
```

Just share a idea.

Reply:
Thanks ! Good point and I'll include that change in a subsequent PR when adding more test cases to the suite.